src/hypertable_restrict_info.c

```diff
@@ -710,7 +702,27 @@ ts_hypertable_restrict_info_get_chunks(HypertableRestrictInfo *hri, Hypertable *
 			}
 			else
 			{
-				chunk_ids = list_append_unique_int(chunk_ids, osm_chunk_id);
+				/*
+				 * At this point the OSM chunk was either:
+				 * 1. added to the list because it has a valid range that agrees with the
+				 * restrictions;
+				 * 2. not added because it has a valid range and it was excluded;
+				 * 3. not added because it has an invalid range and it was excluded.
+				 * If the chunk's range is invalid, only then should we consider adding it,
+				 * otherwise the exclusion logic should have correctly included or excluded it from
+				 * the list. Also, if the range is invalid but the NONCONTIGUOUS flag is not set,
+				 * indicating that the chunk is empty, we don't need to do a scan so we do not add
+				 * it either.
+				 */
+				const Dimension *time_dim = hyperspace_get_open_dimension(ht->space, 0);
+				DimensionSlice *slice =
```

Comment:
This will abort if we don't get a lock i.e if there are concurrent transactions modifying the slice. Is there a reason to get tuple exclusive lock here?

I believe we only need a KeyShare lock and can do something similar to ts_dimension_slice_scan_range_limit
(would be good to wrap this logic in a function and just call the function here).